standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/utils/TestMetaStoreUtils.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hive.metastore.utils;
+
+import org.apache.hadoop.hive.metastore.annotation.MetastoreUnitTest;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+import java.sql.Date;
+import java.sql.Timestamp;
+import java.time.ZoneId;
+import java.util.TimeZone;
+
+import static org.junit.Assert.assertEquals;
+
+@Category(MetastoreUnitTest.class)
+public class TestMetaStoreUtils {
+
+  @Test
+  public void testConvertDateToString() {
+    TimeZone defaultTZ = TimeZone.getDefault();
+    try {
+      TimeZone.setDefault(TimeZone.getTimeZone(ZoneId.of("Asia/Hong_Kong")));
+      String date = MetaStoreUtils.convertDateToString(Date.valueOf("2023-01-01"));
+      assertEquals("2023-01-01", date);
+    } finally {
+      TimeZone.setDefault(defaultTZ);
+    }
+  }
+
+  @Test
+  public void testcConvertTimestampToString() {
+    TimeZone defaultTZ = TimeZone.getDefault();
+    try {
+      String date = MetaStoreUtils.convertTimestampToString(Timestamp.valueOf("2023-01-01 10:20:30"));
+      assertEquals("2023-01-01 10:20:30", date);
+    } finally {
+      TimeZone.setDefault(defaultTZ);
+    }
```

Comment:
Why do we need this? Neither the test nor the called method modify the default timezone.

Reply:
Actually here I missed out  `TimeZone.setDefault(TimeZone.getTimeZone(ZoneId.of("Asia/Hong_Kong")));` line in the try block. Thanks for pointing it out, now added in all the testcases.

Reply:
Since all tests are switching timezone it would be nice to do some refactoring to avoid the boilerplate. Using `@Before`, `@After` annotated methods and/or `@Parameterized` tests could aid in this.

Reply:
Updated the testcases accordingly but I couldn't use `@Parameterized` due to some maven dependency issues.

Reply:
I pushed a commit (https://github.com/apache/hive/pull/4760/commits/4ff9ba968b31ea16611ba5d5bc05bbfb3b4fe184) based on what I had in mind. Have a look and let me know if I missed something.

With the new tests, I found a bug in `convertTimestampToString` using [zoneId=Europe/Paris, timestamp=9743-03-31T03:37:33] but I think this existed since day one so we can possibly defer it to another PR.

Reply:
Yeap I agree its not a problem introduced by the PR but it is a bug that is probably due to DST shift. We have many of those in Hive :) Please log a new JIRA ticket and we can follow-up there.

Reply:
Issue is observed for the timestamps from `9743-03-31 03:00:00` to `9743-03-31 03:59:59`.

Reply:
https://issues.apache.org/jira/browse/HIVE-27775 is the ticket for the problem reported here.